sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/DropTableSuite.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.v1
+
+import org.apache.spark.sql.execution.command
+
+trait DropTableSuiteBase extends command.DropTableSuiteBase {
+  test("purge option") {
+    withNamespace(s"$catalog.ns") {
+      sql(s"CREATE NAMESPACE $catalog.ns")
+
+      createTable(s"$catalog.ns.tbl")
+      checkTables("ns", "tbl")
+
+      sql(s"DROP TABLE $catalog.ns.tbl PURGE")
+      checkTables("ns") // no tables
+    }
+  }
+}
+
+class DropTableSuite extends DropTableSuiteBase with CommandSuiteBase {
+  // The test fails in Hive External catalog with:
+  // org.apache.spark.sql.AnalysisException:
+  //   spark_catalog.ns.tbl is not a valid TableIdentifier as it has more than 2 name parts.
```

Comment:
do you know what goes wrong?

Reply:
I haven't investigated yet why v1 In-Memory and Hive ext catalog have different behavior. The same approach in tests worked for other commands. It seems something specific to `DROP TABLE`. Need to look at it separately since this PR about tests.

Reply:
This is some kind of Hive specific check:
```
  spark_catalog.ns.tbl is not a valid TableIdentifier as it has more than 2 name parts.
  org.apache.spark.sql.AnalysisException: spark_catalog.ns.tbl is not a valid TableIdentifier as it has more than 2 name parts.
      at org.apache.spark.sql.connector.catalog.CatalogV2Implicits$MultipartIdentifierHelper.asTableIdentifier(CatalogV2Implicits.scala:130)
      at org.apache.spark.sql.hive.test.TestHiveQueryExecution$$anonfun$1.applyOrElse(TestHive.scala:606)
      at org.apache.spark.sql.hive.test.TestHiveQueryExecution$$anonfun$1.applyOrElse(TestHive.scala:606)
```